src/forms/Y2021/irsForms/ScheduleE.ts

```diff
@@ -212,6 +238,42 @@ export default class ScheduleE extends Form {
     const tp = new TaxPayer(this.state.taxPayer)
     const [p0, p1, p2] = [0, 1, 2].map((i) => this.propForRow(i))
 
+    const k1s = this.state.scheduleK1Form1065s
+    const l28Fields: Array<string | number | boolean | undefined> = []
+    for (let i = 0; i < k1s.length && i < 4; i++) {
```

Comment:
Also, we need at least a TODO for handling more than 4 k1s. I think TaxAct appends a statement here. We don't have any ability for that but I think if we did it would make things a lot easier in this case and many others.

Reply:
Interesting.  I don't think I've ever used `flatMap` before.  I think I did that right.  Also added a TODO.

Reply:
Totally a style difference, some people can read a for loop easier, for me I like to understand in terms of `map` (same length), `flatMap` (grows or shrinks length), `reduce`/`fold` (accumulate single value).